enhancements/hypershift/hostedClusterInfrastructure.md

```diff
@@ -0,0 +1,139 @@
+
+When a new HostedClusterInfrastructure is created:
+- The HostedClusterInfrastructure controller will reconcile
+  - First it will determine if a Cloud Provider credential is present
+    - If Present:
+      - It will create the infrastructure where needed and validate the infrastructure where it exists
+    - If Not Present
+      - The resource just contains user provided values that can be consumed by the UI or CLI
+- CLI will be able to create or render the HostedClusterInfrastructure resource
+- CLI will be able to consume the HostedClusterInfrastructure resource
+- User may create/apply the resource with all values
+- Delete HostedClusterInfrastructure the controller will reconcile
+  - First it will determine if a Cloud Provider credential is present
+    - If Present:
+      - It will delete all the infrastructure resources found in the cr before allowing the cr to be removed
+    - If Not Present
+      - The cr will just be deleted
+
+### CLI
+Supported CLI capabilities:
+- Create a HostedClusterInfrastructure using an AWS/Azure credential or secret
+- Render a HostedClusterInfrastructure using an AWS/Azure credential or secret (for Git or manual create/apply)
+- Consume an existing HostedClusterInfrastructure when creating a HostedCluster and NodePool resource
+- If connected to the Hosting Cluster(Management Cluster) When creating infrastructure apply the HCI cr as part of the creation process
+
+#### Console
+- Allow creation of infrastructure via the HCI cr
+- Allow deletion of infrastructure via the HCI cr
+- Allow Admin to pre-populate infrastructure for use by a consumer who just creates the HostedCluster and NodePool cr's
+- (spike) Report status of resources in HCI
+### Workflow Description
+The [HostedClusterInfrastructure API](https://github.com/openshift/hypershift/blob/27c0a432bdc8d702f1cdb2a2f3f25e5ae6fbee7d/api/v1alpha1/hostedclusterinfrastructure.go) 
+is the entrypoint and consumer facing API for any human or automation to interact with cloud provider infrastructure.  This works in tandem with the hypershift CLI, but is neither is required to create HostedClusters if you have the appropriate values.
+
+### API Extensions
+N/A.
+### Risks and Mitigations
+N/A.
+### Drawbacks
+N/A.
+### Test Plan
+N/A.
+#### Dev Preview -> Tech Preview
+#### Tech Preview -> GA
+The following features are aimed to be supported and go through e2e automated testing before GA:
+- Create Infra
+- Destroy Infra
+- (Spike) Verify infra
+
+#### Removing a deprecated feature
+- This will align with ACM 2.7, where the HypershiftDeployment CRD will be removed
+### Upgrade / Downgrade Strategy
+N/A.
+### Version Skew Strategy
+N/A.
+### Operational Aspects of API Extensions
+N/A.
+#### Failure Modes
+#### Support Procedures
+
+## Alternatives
+
```

Comment:
Can we elaborate this section?

Reply:
Added some of the alternatives we discussed in Untangled.